internal/pkg/graph/graph.go

```diff
@@ -100,21 +123,67 @@ func (g *Graph[V]) hasCycles(temp *findCycleTempVars[V], currVertex V) bool {
 	return false
 }
 
-// TopologicalSorter ranks vertices in a graph using topological sort.
-type TopologicalSorter[V comparable] struct {
-	ranks map[V]int
+// LevelOrder ranks vertices in a breadth-first search manner.
+type LevelOrder[V comparable] struct {
+	marked map[V]bool
```

Comment:
would it make sense to just create a new map inside `traverse()` each time it's called instead of having it on the struct? then it wouldn't have to get reset [here (L185)](https://github.com/aws/copilot-cli/pull/3497/files#diff-c28ac7314b17a9809b0254ccb9d73bb5ff6dad0081dd0329ac77e85a8f598cd4R185)

Reply:
Thanks! much better, fixed it.